### PR TITLE
[1.18.x] Changing hardcoded Items.FISHING_ROD to allow for custom fishing rods

### PR DESCRIPTION
**Pull request by rehwinkel**
1.18 port of #8168.

### CODE REVIEW

**Comment by pupnewfster** [comment]
Shouldn't these be using forge's [instanceof transformer](https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/build.gradle#L1084)?

**Comment by KnightMiner** [comment]
Alternatively, is this a case that would benefit from tool actions?

Yes I know, repeat of mine and pup's comments from #8168.

**Comment by eerussianguy** [comment]
There are only two relevant patches, one of which is in the renderer hook. I don't think we need to overthink this.

**Comment by KnightMiner** [comment]
Using tool actions instead of instanceof is not about overthinking it, its  about flexibility. instanceof means you prevent anyone from benefitting from this patch unless they extend the vanilla class. Is there a case where someone would want this and not be extending the vanilla class?

See also the shears patches, they use the tool action instead of instanceof shears.

----

The transformer suggested is also not overthinking it. Forge does not want to maintain any patches that are trivial transformers. It is a 3 line "config" file addition, does not need to be a patch.

**Comment by rehwinkel** [comment]
@KnightMiner I implemented it using ToolAction!

File: src/test/java/net/minecraftforge/debug/item/CustomFishingRodTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        public CustomFishingRodItem(Item.Properties props)
+        {
+            super(props);
+        }
+
+        @Override
+        public boolean canPerformAction(ItemStack stack, ToolAction toolAction)
+        {
+            return toolAction.equals(ToolActions.FISHING_ROD_CAST);
+        }
```
Comment (pupnewfster) [inline src/test/java/net/minecraftforge/debug/item/CustomFishingRodTest.java]: This shouldn't be needed as  this class is extending FishingRodItem

File: src/main/java/net/minecraftforge/common/ToolActions.java [hunk 2]
```diff
@@ -114,6 +114,8 @@
      */
     public static final ToolAction SHIELD_BLOCK = ToolAction.get("shield_block");
 
+    public static final ToolAction FISHING_ROD_CAST = ToolAction.get("fishing_rod_cast");
```
Comment (pupnewfster) [inline src/main/java/net/minecraftforge/common/ToolActions.java:120]: This should probably have a one sentence javadoc like the other tool actions do.

**Comment by GirafiStudios** [comment]
@rehwinkel looks like this PR unfortunately have some conflicts now. Could you resolve those? :)

**Comment by mcenderdragon** [comment]
um any news on this ?

**Comment by sciwhiz12** [comment]
Hello! @rehwinkel, please leave a comment whether you wish to continue this PR. As a similar PR was merged to 1.19.2 recently -- #8681 -- this should conform to the changes in that PR, or an explanation be given on why it should differ in implementation. If we do not hear back from you after some time, this PR will be closed for inactivity.

If anyone else wishes to continue the work of this PR, please create a PR with a link to this PR for reference.
Thank you for your contributions. 👋

**Comment by rehwinkel** [comment]
This PR can be closed, I'm not going to be working on this any time soon.

**Comment by sciwhiz12** [comment]
Closing for now. Thank you for your contributions!

**Pull request closed without merging**